test/string.c

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright © 2017-2018 The Crust Firmware Authors.
+ * SPDX-License-Identifier: (BSD-3-Clause OR GPL-2.0)
+ */
+
+#include <stddef.h>
+#include <string.h>
+#include <unity.h>
+
+static void
+test_Strcmp(void)
+{
+    /* Test with long string. */
+    TEST_ASSERT_EQUAL(0, strcmp("Here is a string", "Here is a string"));
+
+    /* Test with medium string. */
+    TEST_ASSERT_EQUAL(0, strcmp("Hello World", "Hello World"));
+
+    /* Test with short string. */
+    TEST_ASSERT_EQUAL(0, strcmp("A", "A"));
+
+    /* Test with numbers in string. */
+    TEST_ASSERT_EQUAL(0, strcmp("He11o W0rld", "He11o W0rld"));
+
+    /* Test with special characters. */
+    TEST_ASSERT_EQUAL(0, strcmp("Hello_World!", "Hello_World!"));
+
+    /* Test string with only a space. */
+    TEST_ASSERT_EQUAL(0, strcmp(" ", " "));
+
+    /* Test an empty string. */
+    TEST_ASSERT_EQUAL(0, strcmp("", ""));
+
+    /* Test two different strings. */
+    int result = strcmp("Here is a string", "Here is a different string");
+    TEST_ASSERT(result != 0);
```

Comment:
The result being positive or negative is meaningful here (you should have tests for both).

Reply:
done